File: 1545-find-kth-bit-in-nth-binary-string/1545-find-kth-bit-in-nth-binary-string.py
class Solution:
    def findKthBit(self, n: int, k: int) -> str:
        def find(n: int) -> str:
            if n == 1:
                return "0"

            before = find(n - 1)
            after = list()

            for bit in before:
                if bit == "1":
                    after.append("0")
                else:
                    after.append("1")

            return before + "1" + "".join(after[::-1])

        result = find(n)
        print(result)
        return result[k - 1]

File: 1545-find-kth-bit-in-nth-binary-string/1545-find-kth-bit-in-nth-binary-string.java
class Solution {
    private StringBuilder result;

    private void find(int n) {
        if (n == 1) {
            result.append("0");
            return;
        }

        find(n - 1);

        StringBuilder sb = new StringBuilder();
        sb.append("1");

        for (int i = result.length() - 1; i >= 0; i--) {
            if (result.charAt(i) == '0') {
                sb.append("1");
            } else {
                sb.append("0");
            }
        }

        result.append(sb);
    }

    public char findKthBit(int n, int k) {
        result = new StringBuilder();
        find(n);

        return result.charAt(k - 1);
    }
}